components/brave_wallet/browser/keyring_service.cc

```diff
@@ -1681,7 +1723,11 @@ void KeyringService::SetKeyringDerivedAccountName(
 bool KeyringService::UpdateNameForHardwareAccountSync(
     const std::string& address,
     const std::string& name) {
-  base::Value* hardware_keyrings = GetPrefForHardwareKeyringUpdate(prefs_);
+  auto keyring_id = GetKeyringIdForHardwareAccount(address);
```

Comment:
We should update the API `SetDefaultKeyringHardwareAccountName` to `SetHardwareAccountName(string address, string name, CoinType coin)`.  `CoinType` info are available in `WalletInfo` hence we don't need to do iteration to find keyring here

Reply:
done

Reply:
This function should also takes `CoinType from` `SetHardwareAccountName` so we can get keyring id from it

Reply:
done